src/main/java/eu/luminis/elastic/cluster/ClusterManagementService.java

```diff
@@ -0,0 +1,110 @@
+package eu.luminis.elastic.cluster;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.http.Header;
+import org.apache.http.HttpHost;
+import org.apache.http.message.BasicHeader;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.sniff.Sniffer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import eu.luminis.elastic.LoggingFailureListener;
+
+@Component
+public class ClusterManagementService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClusterManagementService.class);
+    private static final String HEADER_CONTENT_TYPE_KEY = "Content-Type";
+    private static final String DEFAULT_HEADER_CONTENT_TYPE = "application/json";
+
+    private LoggingFailureListener loggingFailureListener;
+    private Map<String, Cluster> clusters = new HashMap<>();
+    private Cluster currentCluster;
+
+    @Value("${enableSniffer:true}")
+    private boolean enableSniffer = true;
+
+    @Autowired
+    public ClusterManagementService(LoggingFailureListener loggingFailureListener) {
+        this.loggingFailureListener = loggingFailureListener;
+    }
+
+    public Optional<Cluster> getCurrentCluster() {
+        return (currentCluster == null) ? Optional.empty() : Optional.of(currentCluster);
+    }
+
+    public Cluster setCurrentCluster(String clusterName) {
+        Cluster cluster = clusters.get(clusterName);
+        if(cluster != null) {
+            currentCluster = cluster;
+        }
+        return currentCluster;
+    }
+
+    public Cluster addCluster(String clusterName, List<String> hosts) {
+        if(!exists(clusterName)) {
+            Cluster cluster = createCluster(hosts);
+            clusters.put(clusterName, cluster);
+            return cluster;
+        } else {
+            return null;
+        }
+    }
+
+    public Cluster deleteCluster(String clusterName) {
+        return closeCluster(clusterName);
+    }
+
+    @Override
+    protected void finalize() {
```

Comment:
I don't like this Override of finalise, prefer the @PreDestroy tag to only destroy it in a spring context, we will not really use the other method as it is a code smell.